src/doc/trpl/installing-rust.md

```diff
@@ -26,48 +26,108 @@ the [64-bit installer][win64] and run it.
+Rust, Cargo, and multiple custom toolchains; mappable to any working directory. Multirust is still a work in progress
+and is not a part of Rust:master.(Yet)
+
+Rust is signed with a [PGPKey](../trpl/pgp.key)!
+
+To install 'multirust' from Github first move to your project directory.
+
+````bash
+$ cd /home/$usr/rust/
+```
+Next we'll want to get the sourcecode from Github.
+
+```bash
+$ git clone --recursive https://github.com/brson/multirust
+```
+
+We will then move to the new multirust directory.
+
+```bash
+$ cd ./rust/multirust
+```
+
+Now we will initialize the submodule
+
+```bash
+$ git submodule update --init
+```
+
+And finally we will install 'multirust' which will handle the installation of Rust and Cargo!
+(This step will require your root password)
+
+```bash
+./build.sh && sudo ./install.sh
+```
+See [multirust](../doc/trpl/multirust_advsetup.md) advanced setup for toolchain configuration options and everything you ever wanted to know and more about multirust.
+
+Otherwise if you are working on your own machine and trust the development team.
+
+Install multirust via
+
+```bash
+$ sudo curl -sf https://raw.githubusercontent.com/brson/multirust/master/blastoff.sh | sh
+```
+Or Rust proper (minus multirust)
+
+```bash
+$ curl -sf -L https://static.rust-lang.org/rustup.sh | sudo sh
+```
+
 ## Uninstalling
 
-If you decide you don't want Rust anymore, we'll be a bit sad, but that's okay.
-Not every programming language is great for everyone. Just run the uninstall
-script:
+If you decide you don't want Rust anymore, we'll make sure to keep your seat warm until you get back.
+
+If you used the Windows installer, just re-run the `.msi` and it will give you an uninstall option.
+
+If you maintain your Rust installation via multirust uninstall is simple.
+
+```bash
+$ sudo ./install.sh --uninstall
+```
+Note: If you added additional install options as described on the [multirust](../doc/trpl/multirust_advsetup.md)
+page you will want to see that same page for the additional uninstallation options.
+
+If you did not use multirust the command is 
 
 ```bash
 $ sudo /usr/local/lib/rustlib/uninstall.sh
 ```
 
-If you used the Windows installer, just re-run the `.msi` and it will give you
-an uninstall option.
+##Installation test and Upgrade!
 
-Some people, and somewhat rightfully so, get very upset when we tell you to
-`curl | sudo sh`. Basically, when you do this, you are trusting that the good
-people who maintain Rust aren't going to hack your computer and do bad things.
-That's a good instinct! If you're one of those people, please check out the
-documentation on [building Rust from Source][from source], or [the official
-binary downloads][install page]. And we promise that this method will not be
-the way to install Rust forever: it's just the easiest way to keep people
-updated while Rust is in its alpha state.
+Upgrading to the latest nightly is easy!
```

Comment:
We put nightly instructions in the unstable section of the guide, so this shoulnd't be here